src/common/tracer.cc

```diff
@@ -140,4 +143,26 @@ EmptySpanID(const TraceContext* ctx) {
     return isEmptyID(ctx->spanID, trace::SpanId::kSize);
 }
 
+std::string
+StringToHex(const std::string& input) {
+    std::stringstream ss;
+    ss << std::hex << std::setfill('0');
+    for (unsigned char c : input) {
+        ss << std::setw(2) << static_cast<int>(c);
+    }
+    return ss.str();
+}
+
+std::string
+HexToString(const std::string& input) {
+    std::string output;
+    std::stringstream ss;
+    for (size_t i = 0; i < input.length(); i += 2) {
+        std::string byteString = input.substr(i, 2);
+        char byte = static_cast<char>(std::stoi(byteString, nullptr, 16));
+        output += byte;
```

Comment:
`ss` is not used, so it seems that you originally wanted to do something like `ss << byte;`
So, please either remove `ss` or use it instead of `output += byte;`

Reply:
updated